qiskit_nature/problems/second_quantization/vibrational/vibrational_op_builder.py

```diff
@@ -57,6 +57,25 @@ def build_vibrational_op(watson_hamiltonian: WatsonHamiltonian,
     boson_hamilt_harm_basis = HarmonicBasis(watson_hamiltonian,
                                             num_modals, truncation_order).convert()
 
-    all_labels = _create_labels(boson_hamilt_harm_basis)
+    return build_vibrational_op_from_ints(boson_hamilt_harm_basis, num_modes, num_modals)
+
+
+def build_vibrational_op_from_ints(h_mat: List[List[Tuple[List[List[int]], complex]]],
+                                   num_modes: int,
+                                   num_modals: List[int],
+                                   ) -> VibrationalOp:
+    """
+    Builds a :class:`VibrationalOp` based on an integral list as produced by
+    :meth:`HarmonicBasis.convert()`.
```

Comment:
Just an FYI: I used this code basically ad verbatim from Aqua. I am not sure how this will generalize once we have bosonic bases other than the harmonic one.